### PR TITLE
feat(examples): add Base UI integration

### PR DESCRIPTION
**Pull request by edmundhung**
## Summary

- add a pure @base-ui/react example using typed configureForms metadata
- keep Base UI native and hidden inputs canonical, with useControl only for the controlled Slider
- cover validation focus and blur, FormData serialization, URL defaults, Conform and browser resets, and reset-to-submitted values
- align the Vite, TypeScript, Oxlint, README, Playwright, and pnpm injection setup with the recently modernized examples

## Verification

- pnpm --filter base-ui-example run build
- pnpm --filter base-ui-example run lint
- pnpm --filter base-ui-example run typecheck
- pnpm --filter base-ui-example run test (27 passed across Chromium, Firefox, and WebKit)
- pnpm exec oxfmt --check examples/base-ui
- frozen lockfile install and git diff checks

<details>
<summary>Generated Summary</summary>

- Added a standalone `@base-ui/react` example with typed `configureForms` metadata.
- Demonstrated native and hidden inputs, controlled Slider integration, validation, focus and blur handling, URL defaults, form resets, and reset-to-submitted values.
- Added reusable accessible field components, styling, documentation, and 27 Playwright tests across Chromium, Firefox, and WebKit.
- Added Vite, TypeScript, Oxlint, Playwright, and pnpm dependency configuration for the example.

</details>

### CODE REVIEW

File: examples/base-ui/src/components.tsx [hunk 1]
```diff
@@ -0,0 +1,485 @@
...
+export function SliderField(props: StringFieldProps) {
+	const { id, label, invalid, name, defaultValue } = props;
+	const thumbInputRef = useRef<HTMLInputElement>(null);
+	const control = useControl({
+		defaultValue: defaultValue ?? '50',
+		onFocus() {
+			thumbInputRef.current?.focus();
+		},
+	});
+	const value = Number(control.value ?? 50);
```
[the remark this answers is not in the document]
Comment (edmundhung) [reply examples/base-ui/src/components.tsx]: Fixed in d866703f. Both Slider fallback points now treat an empty string as missing, and the Playwright numeric-control test covers an explicitly empty budget URL value.

File: examples/base-ui/README.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
+# Base UI Example
+
+> This guide focuses on behavior specific to Base UI. See [Integrating with UI Libraries](../../docs/integration/ui-libraries.md) for the general concept and the [`useControl`](../../docs/api/react/future/useControl.md) API.
+
+[Base UI](https://base-ui.com/) provides unstyled, accessible React primitives. This example integrates Conform directly with `@base-ui/react`. Unlike [`examples/shadcn-base-ui`](../shadcn-base-ui), it does not use shadcn/ui components, generated registries, or copied shadcn code.
```
[the remark this answers is not in the document]
Comment (edmundhung) [reply examples/base-ui/README.md]: Fixed in d866703f. The sibling example is not on main yet, so the required comparison now renders examples/shadcn-base-ui as code instead of a broken relative link.

File: examples/base-ui/src/components.tsx [hunk 3]
```diff
@@ -0,0 +1,573 @@
...
+export function NumberFieldControl(props: StringFieldProps) {
+	const { id, label, invalid, name, defaultValue } = props;
+	const inputRef = useRef<HTMLInputElement>(null);
+	const control = useControl({
+		defaultValue,
+		onFocus() {
+			inputRef.current?.focus();
+		},
+	});
+	const value = control.value ? Number(control.value) : null;
```
[the remark this answers is not in the document]
Comment (edmundhung) [reply examples/base-ui/src/components.tsx]: Fixed in 589c1a6a. Note that the string `"0"` was already truthy in JavaScript, so the previous expression preserved it; the updated conversion now checks the empty string explicitly and maps `NaN` to `null`, making both cases unambiguous. Build, lint, typecheck, and all six Playwright projects pass.

File: examples/base-ui/src/components.tsx [hunk 4]
```diff
@@ -0,0 +1,572 @@
...
+export function SliderField(props: StringFieldProps) {
+	const { id, label, invalid, name, defaultValue } = props;
+	const thumbInputRef = useRef<HTMLInputElement>(null);
+	const control = useControl({
+		defaultValue: defaultValue || '50',
+		onFocus() {
+			thumbInputRef.current?.focus();
+		},
+	});
+	const value = Number(control.value || 50);
```
[the remark this answers is not in the document]
Comment (edmundhung) [reply examples/base-ui/src/components.tsx]: Fixed in a41e8399. Slider values now fall back to 50 whenever the URL-derived numeric value is non-finite, and the validation lifecycle test covers both `budget=` and `budget=abc`. Build, lint, typecheck, formatting, and all six Playwright projects pass.

File: examples/base-ui/src/components.tsx [hunk 5]
```diff
@@ -0,0 +1,573 @@
...
+	const { id, label, invalid, name, defaultValue } = props;
+	const thumbInputRef = useRef<HTMLInputElement>(null);
+	const control = useControl({
+		defaultValue: defaultValue || '50',
+		onFocus() {
+			thumbInputRef.current?.focus();
+		},
+	});
+	const numericValue = Number(control.value);
+	const value = Number.isFinite(numericValue) ? numericValue : 50;
```
[the remark this answers is not in the document]
Comment (edmundhung) [reply examples/base-ui/src/components.tsx:362]: No change needed here. Base UI 1.7 already clamps controlled Slider values internally: `SliderRoot` derives its rendered `values` with `clamp(valueUnwrapped, min, max)` before calculating the thumb/indicator state. The separate named `BaseControl` intentionally retains the raw URL value so Conform/Zod can report `budget=105` or `budget=-5` as invalid. Clamping in this adapter would duplicate Base UI behavior; only non-finite values need normalization before reaching the primitive.
